chunky/src/java/se/llbit/chunky/world/Chunk.java

```diff
@@ -188,7 +188,7 @@ private void loadSurface(Map<String, Tag> data, ChunkData chunkData) {
         loadBlockData(data, chunkData, palette);
         int[] heightmapData = extractHeightmapData(data, chunkData);
         updateHeightmap(heightmap, position, chunkData, heightmapData, palette);
```

Comment:
```suggestion
        updateHeightmap(heightmap, position, chunkData, heightmapData, palette, yMax);
```
and in implementation change `Math.max(1, y - 1)` to `Math.max(1, Math.min(y - 1, yMax)`